builtin/logical/pki/path_root.go

```diff
@@ -257,13 +257,16 @@ func (b *backend) pathCAGenerateRoot(ctx context.Context, req *logical.Request,
 
 	// Also store it as just the certificate identified by serial number, so it
 	// can be revoked
+	b.certsCountedLock.RLock() // Don't allow storage of certs during initial count
 	err = req.Storage.Put(ctx, &logical.StorageEntry{
 		Key:   "certs/" + normalizeSerial(cb.SerialNumber),
 		Value: parsedBundle.CertificateBytes,
 	})
+	b.certsCountedLock.RUnlock() // Unlock before we build the CRL
 	if err != nil {
 		return nil, fmt.Errorf("unable to store certificate locally: %w", err)
 	}
+	b.incrementTotalCertificatesCount()
```

Comment:
I'm not really inclined to increment here for generated roots, tbh. While in the storage, they're mostly exposed there for $LEGACY_REASONS aren't traditional certs that you'd want to cleanup (until they expire...). Plus, they're duplicated into Issuers storage which is the real "canonical" source if you would.

Reply:
It doesn't look like, other than reading/parsing the cert, there's an easy way to tell whether a stored cert is a root or leaf.  So, if we removed this, I'd be worried that someone will see 'cool no certs', then turn the node on-and-off 'where did these nine certs come from!?' (when the list count-completes).

Checking: it looks like these currently appear in tidy-counts too, so if we removed them, the tidy-count could be higher than the certs-in-storage too.  I think that would be more confusing.

Reply:
Ahhhh yeah, good points :-) I'll retract this suggestion then. :D